.github/workflows/test.yaml

```diff
@@ -18,9 +18,9 @@ jobs:
     - uses: actions/checkout@v3
 
     - name: Set up Go
-      uses: actions/setup-go@v3
+      uses: actions/setup-go@v4
       with:
-        go-version: 'stable'
+        go-version-file: 'go.mod'
```

Comment:
This means we'll need to update go.mod at release to ensure we don't release using Go with a CVE though, right?

Reply:
We have `check-latest: true` so it will only be used to bump minor release, which we'll probably do via `go.mod`, I assume :thinking:

Reply:
Sure, I just think it's a nice mechanism testing what the codebase is set to run with, but there could be value in having it future tested too.